crates/cheatcodes/spec/src/vm.rs

```diff
@@ -499,43 +497,61 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Unsafe)]
     function readCallers() external returns (CallerMode callerMode, address msgSender, address txOrigin);
 
+    // ----- Arbitrary Snapshots -----
+
+    // TODO: add group support
+
+    /// Snapshot capture an arbitrary numerical value by name.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function snapshotValue(string calldata name, uint256 value) external returns (bool success);
+
+    // -------- Gas Snapshots --------
+
+    /// Start a snapshot capture of the current gas usage by name.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function startSnapshotGas(string calldata name) external;
+
+    /// Stop the snapshot capture of the current gas usage by name, capturing the gas used since the start.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function stopSnapshotGas(string calldata name) external returns (bool success, uint256 gasUsed);
+
     // -------- State Snapshots --------
 
     /// Snapshot the current state of the evm.
     /// Returns the ID of the snapshot that was created.
     /// To revert a snapshot use `revertTo`.
     #[cheatcode(group = Evm, safety = Unsafe)]
-    function snapshot() external returns (uint256 snapshotId);
+    function snapshotState() external returns (uint256 snapshotId);
```

Comment:
I agree with this set of name change for state snapshots, since the term "snapshot" is now broad, but just noting this is a breaking change so we'll need to make sure to communicate that

Reply:
We can should the old names and mark them as deprecated (`#[cheatcode(... status = Deprecated)]`, <https://github.com/foundry-rs/foundry/blob/cb109b1699f82d009574d13aa59f1585a3fbfdb2/crates/cheatcodes/spec/src/cheatcode.rs#L48>), these kinds of breaking changes are not a good experience.

I don't believe the deprecated warnings logic is implemented yet, but it should be straight forward to implement. CC @mattsse @klkvr

Reply:
Re-added the old cheatcodes and tagged them as `Deprecated`, sharing the inner logic with the new implementation